controllers/user.js

```diff
@@ -27,24 +28,110 @@ router.get( '/:id', (req, res, next) => {
 
 
 router.put( '/:id', (req, res, next) => {
-  req.user.update( req.body, (error, result) => {
+  // mask data (only pass on qualifying values)
+  const updateData = {};
```

Comment:
You are accessing to many times the `req.body` object. Try to destruct it and then apply the logic.